From f16cd8c4986b7f250e8465dfbe4bda8d366e05dd Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex@linutronix.de>
Date: Mon, 4 Mar 2024 14:35:16 +0100
Subject: [PATCH] man.ml: do not use /dev/stdin

This was introduced in https://git.kernel.org/pub/scm/docs/man-pages/man-pages.git/commit/?id=30c38a8bf8ae8f4a6e71d3b8ac4abf0a40778f1e
(presumably to build a 'sed | install' pipeline instead of previous 'install; sed' sequence),
but it doesn't work under pseudo where /dev/stdin is absent.

Upstream-Status: Inappropriate [yocto specific]
Signed-off-by: Alexander Kanavin <alex@linutronix.de>
---
 share/mk/install/man.mk | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/share/mk/install/man.mk b/share/mk/install/man.mk
index b9865b9..06880f0 100644
--- a/share/mk/install/man.mk
+++ b/share/mk/install/man.mk
@@ -60,10 +60,10 @@ $(foreach s, $(MANSECTIONS),                                                  \
 
 $(_manpages):
 	$(info INSTALL	$@)
-	<$< \
-	$(SED) $(foreach s, $(MANSECTIONS), \
+	$(INSTALL_DATA) -T $< $@
+	$(SED) -i $(foreach s, $(MANSECTIONS), \
 		-e '/^\.so /s, man$(s)/\(.*\)\.$(s)$$, $(notdir $(man$(s)dir))/\1$(man$(s)ext)$(Z),') \
-	| $(INSTALL_DATA) -T /dev/stdin $@
+	$@
 ifeq ($(LINK_PAGES),symlink)
 	if $(GREP) '^\.so ' <$@ >/dev/null; then \
 		$(GREP) '^\.so ' <$@ \
